cEP-0016.md

```diff
@@ -0,0 +1,180 @@
+# Git Commit Content Inspection
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 16                                                |
+| Version  | 1.0                                               |
+| Title    | Git commit content inspection                     |
+| Authors  | Kriti Rohillla <mailto:kritirohilla567@gmail.com> |
+| Status   | Implementation due                                |
+| Type     | Process                                           |
+
+## Abstract
+
+This document describes coala process rules which will be enforced by the
+GitCommitBear. Some new changes will be introduced in the implementation of
+GitCommitBear based on which the commits made to coala will be detected and
+verified.
+
+## Introduction
+
+The GitCommitBear at present performs a check on the content of regular commits
+made to coala. However, there are many special types of commit messages that
+should be used only in conjunction with patches containing a special type of
+content.
+
+For example, git revert commit which is used to revert any previous commit and
+git merge commit which is generated whenever someone performs a git rebase and
+tries to merge master onto the pull request. Such commits generate a default
+commit message that is very concise and not according to coala’s standards.
+The coala project wants to allow `git revert` commits, but does not want to
+accept `git merge` commits as the repositories are strictly fast-forward merges
+that do not have an extra commit.
+
+Also in cases where some tests are not required by any pull request made to
+coala, some special sequences can be added to git commit messages to instruct
+tools to operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in .coafile
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely revert,
+   merge and commits that disable ci builds.
+2. Each type of commit is then handled as a separate case.
+3. For commits that disable ci builds:
+   - Commit contents are checked to find `[ci skip]`.
```

Comment:
Which CI engines does coala use?
Which CI engines will your project support?
What mechanisms does each CI engine use to allow CI to be skipped?

Reply:
> Not done yet.

I am referring to the checks GitCommitBear already performs on the shortlog and commit body. What is not done?

Reply:
If a CI engine doesnt support a ci skip command, the bear could emit an info message?

The other very important CI engine is GitLab CI.

We also use Semaphore CI.

The bear needs to have a dictionary of CI engines and metadata about each, like it has for git hosting providers.